### PR TITLE
UPSTREAM: 58466: tolerate more than one gvklist item

### LINKED ISSUE #17872: [regression] oc explain doesn't work
`oc explain` doesn't work anymore on the latest version:

```console
$ oc explain Pod
Error from server (Forbidden): unknown
```

##### Version
oc v3.9.0-alpha.0+1c24d18-233-dirty
kubernetes v1.8.1+0d5291c
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://10.34.129.148:8443
openshift v3.9.0-alpha.0+1c24d18-233-dirty
kubernetes v1.8.1+0d5291c

##### Steps To Reproduce
1. `oc cluster up --version=latest`
2. `oc explain Pod`

##### Current Result
```console
$ oc explain Pod --loglevel=8
I1219 11:28:07.046240   27763 loader.go:357] Config loaded from file /home/coder/.kube/config
I1219 11:28:07.058325   27763 round_trippers.go:414] GET https://10.34.129.148:8443/swagger-2.0.0.pb-v1
I1219 11:28:07.058338   27763 round_trippers.go:421] Request Headers:
I1219 11:28:07.058346   27763 round_trippers.go:424]     Accept: application/json, */*
I1219 11:28:07.058353   27763 round_trippers.go:424]     User-Agent: oc/v1.8.1+0d5291c (linux/amd64) kubernetes/0d5291c
I1219 11:28:07.058359   27763 round_trippers.go:424]     Authorization: Bearer zMHaSA8jdJtUPX9cDEPdT9TSbzXqRsQSaZxXCQYi-eQ
I1219 11:28:07.068940   27763 round_trippers.go:439] Response Status: 403 Forbidden in 10 milliseconds
I1219 11:28:07.068962   27763 round_trippers.go:442] Response Headers:
I1219 11:28:07.068973   27763 round_trippers.go:445]     Content-Type: application/json
I1219 11:28:07.068980   27763 round_trippers.go:445]     X-Content-Type-Options: nosniff
I1219 11:28:07.068989   27763 round_trippers.go:445]     Content-Length: 260
I1219 11:28:07.068996   27763 round_trippers.go:445]     Date: Tue, 19 Dec 2017 10:28:07 GMT
I1219 11:28:07.069003   27763 round_trippers.go:445]     Cache-Control: no-store
I1219 11:28:07.069081   27763 request.go:873] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"forbidden: User \"developer\" cannot get path \"/swagger-2.0.0.pb-v1\": User \"developer\" cannot \"get\" on \"/swagger-2.0.0.pb-v1\"","reason":"Forbidden","details":{},"code":403}
I1219 11:28:07.069282   27763 helpers.go:201] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "unknown",
  "reason": "Forbidden",
  "details": {
	"causes": [
	  {
		"reason": "UnexpectedServerResponse",
		"message": "unknown"
	  }
	]
  },
  "code": 403
}]
F1219 11:28:07.069316   27763 helpers.go:119] Error from server (Forbidden): unknown
$ oc policy can-i get /swagger-2.0.0.pb-v1
no
```

##### Expected Result
Pod schema should be displayed.

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes #17872
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1536845
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1529447

Origin resources, like deployment configs, are part of two api groups -
the legacy "empty" group, and, in cases like deploymentconfigs, the
"apps.openshift.io" group.

**Before**
```
$ oc explain dc
error: Couldn't find resource for "/v1, Kind=DeploymentConfig"
```

**After**
```
$ oc explain dc
DESCRIPTION:
     Deployment Configs define the template for a pod and manages deploying new
     images or configuration changes. A single deployment configuration is
     usually analogous to a single micro-service. Can support many different
     deployment patterns, including full restart, customizable rolling updates,
...
```

cc @deads2k @soltysh

### CODE REVIEW

**Comment by juanvallejo** [comment]
Updated to keep on par with changes in upstream PR

**Comment by juanvallejo** [comment]
@soltysh upstream has merged, ptal
https://github.com/kubernetes/kubernetes/pull/58466

File: test/cmd/explain.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+trap os::test::junit::reconcile_output EXIT
+
+# Cleanup cluster resources created by this test
+(
+  set +e
```
Comment (php-coder) [inline test/cmd/explain.sh]: Why `set +e` if we it can be written as:
```bash
oc delete all,templates --all || :
```
?

File: test/cmd/explain.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  oc delete all,templates --all
+  exit 0
+) &>/dev/null
+
+
+os::test::junit::declare_suite_start "cmd/explain"
```
Comment (php-coder) [inline test/cmd/explain.sh]: Use single quotes here, for consistency with other places.

File: test/cmd/explain.sh (hunk 1 above)
Comment (juanvallejo) [reply test/cmd/explain.sh]: I followed the pattern seen in other tests: https://github.com/openshift/origin/blob/master/test/cmd/builds.sh#L7

File: test/cmd/explain.sh (hunk 2 above)
Comment (juanvallejo) [reply test/cmd/explain.sh]: We seem to use double quotes in other tests [1][2][3][4]

1. https://github.com/openshift/origin/blob/master/test/cmd/builds.sh#L16
2. https://github.com/openshift/origin/blob/master/test/cmd/annotations.sh#L13
3. https://github.com/openshift/origin/blob/master/test/cmd/deployments.sh#L13
4. https://github.com/openshift/origin/blob/master/test/cmd/env.sh#L5

**Pull request merged**